nannyml_cloud_sdk/experiment/experiment.py

```diff
@@ -0,0 +1,304 @@
+                    } for metric, config in metrics_configuration.items()]
+                }
+            },
+        })['create_experiment']
+
+    @classmethod
+    def delete(cls, experiment_id: str) -> None:
+        """Delete an experiment.
+
+        Args:
+            experiment_id: ID of the experiment to delete.
+        """
+        execute(_DELETE_EXPERIMENT, {'id': int(experiment_id)})
+
+    @classmethod
+    def add_experiment_data(cls, experiment_id: str, data: pd.DataFrame) -> None:
+        """Add evaluation data to an experiment.
+
+        Args:
+            experiment_id: ID of the experiment.
+            data: Data to be added.
+
+        Note:
+            This method does not update existing data. It only adds new data. If you want to update existing data,
+            use [upsert_data][nannyml_cloud_sdk.experiment.Experiment.upsert_experiment_data] instead.
+        """
+        data_source = cls._get_experiment_data_source(experiment_id)
+        execute(_ADD_DATA_TO_DATA_SOURCE, {
+            'input': {
+                'id': int(data_source['id']),
+                'storageInfo': Data.upload(data),
+            },
+        })
+
+    @classmethod
+    def upsert_experiment_data(cls, experiment_id: str, data: pd.DataFrame) -> None:
+        """Add or update analysis data for an experiment.
+
+        Args:
+            experiment_id: ID of the model.
+            data: Data to be added/updated.
+
+        Note:
+            This method compares existing data with the new data to determine which rows to update and which to add.
+            If you are certain you are only adding new data, it is recommended to use
+            [add_experiment_data][nannyml_cloud_sdk.experiment.Experiment.add_experiment_data]
+            instead for better performance.
+        """
+        data_source = cls._get_experiment_data_source(experiment_id)
+        execute(_UPSERT_DATA_IN_DATA_SOURCE, {
+            'input': {
+                'id': int(data_source['id']),
+                'storageInfo': Data.upload(data),
+            },
+        })
+
+    @classmethod
+    def get_data_history(cls, experiment_id: str) -> List[DataSourceEvent]:
+        """Get the data history for an experiment.
+
+        Args:
+            experiment_id: ID of the experiment.
+
+        Returns:
+            List of events related to reference data for the experiment.
+        """
+        return execute(_GET_EXPERIMENT_DATA_HISTORY, {
+            'experimentId': int(experiment_id),
+        })['evaluation_model']['referenceDataSource']['events']
+
+    @staticmethod
+    @functools.lru_cache(maxsize=128)
+    def _get_experiment_data_source(
+            experimentId: str, filter: Optional[DataSourceFilter] = None
+    ) -> DataSourceSummary:
+        """Get data sources for a model"""
+        return execute(_GET_EXPERIMENT_DATA_SOURCES, {
+            'experimentId': int(experimentId),
+            'filter': filter,
```

Comment:
I think this will error out because the mutation doesn't have a filter parameter. Not sure though, would also expect this came up during testing 🤔